src/configuration.js

```diff
@@ -79,47 +82,75 @@ export class Configuration {
   }
 
   getRules() {
-    var rules = defaultRules;
+    var rules = this.getAllRules();
     var specifiedRules;
 
     if (this.options.rules && this.options.rules.length > 0) {
       specifiedRules = this.options.rules.map(toUpperCamelCase);
-      rules = rules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) >= 0;
       });
     }
 
     // DEPRECATED - This code should be removed in v1.0.0.
     if (this.options.only && this.options.only.length > 0) {
       specifiedRules = this.options.only.map(toUpperCamelCase);
-      rules = defaultRules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) >= 0;
       });
     }
 
     // DEPRECATED - This code should be removed in v1.0.0.
     if (this.options.except && this.options.except.length > 0) {
       specifiedRules = this.options.except.map(toUpperCamelCase);
-      rules = defaultRules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) == -1;
       });
     }
 
     return rules;
   }
 
+  getAllRules() {
+    if (this.rules !== null) {
+      return this.rules;
+    }
+
+    this.rules = [];
+    this.rulePaths.map(rulePath => {
+      this.rules = this.rules.concat(this.loadRules(rulePath));
+    });
+
+    return this.rules;
+  }
+
+  loadRules(dir) {
+    var rules = [];
+
+    readdirSync(dir).forEach(file => {
+      if (path.extname(file) !== '.js') {
+        return;
+      }
+      var rule = require(path.join(dir, file));
+
```

Comment:
I wonder if there's any basic validation that we can perform on `rule` just to make sure people don't accidentally try to load garbage and end up with a weird crash.

For example we could assert that `rule.length == 1` since rule functions should only accept one argument `context`.

Do you think this is overkill?

Reply:
Yeah, would be great to have. I considered and punted on it at the time, but I could foresee that it might be a poor developer experience as you also noted. Ideally we can add it into the `validate` method and get all that goodness, I'll take a look. If it doesn't seem easy enough to add, I think worse case we can fail hard here with a nicer message.

Reply:
Yeah, I wouldn't worry too much about this, we can still 🚢 as is once we address the other comment. 😀

Reply:
Ok, last commit only addressed the other comment. It might take a bit of restructuring to be able to surface more meaningful error messages at this point.